src/filter/opt.rs

```diff
@@ -418,6 +418,14 @@ pub fn invert(filter: Filter) -> JoshResult<Filter> {
         Op::Subdir(path) => Some(Op::Prefix(path)),
         Op::File(path) => Some(Op::File(path)),
         Op::Prefix(path) => Some(Op::Subdir(path)),
+        Op::SubtreePrefix {
+            subtree_tip: _,
+            path: _,
+        } => {
+            // We assume that new commits being added don't need the prefix fixup any more.
+            // FIXME(RalfJung): does that make sense?
```

Comment:
The alternative would be to do the ancestor test as well here. Not sure if that makes sense though, it would only fire if the commits line up perfectly -- I don't quite know what even the commit ID that gets passed down here. (Or rather, currently it doesn't get passed down yet, the callers would have to propagate that.)